### PR TITLE
add macOS compatibility

### PR DESCRIPTION
**Pull request by crclark96**
minor changes to utility differences in macOS vs Linux
addresses #1

### CODE REVIEW

File: pwned_pass.sh [hunk 1]
```diff
@@ -29,7 +29,12 @@ if [ -z "$password" ]; then
 fi
 set -u
 
-hash=$(printf "%s" "$password" | openssl sha1 | awk '{print $2}')
+if [ $(uname) == "Linux" ]; then
+    hash=$(printf "%s" "$password" | openssl sha1 | awk '{print $2}')
+elif [ $(uname) == "Darwin" ]; then
+    hash=$(printf "%s" "$password" | openssl sha1 | awk '{print $1}')
```
Comment (codyps) [inline pwned_pass.sh]: This breaks compatibility with folks that have openssl from ports installed. I'd recommend instead just taking the last word:

```
 hash=$(printf "%s" "$password" | openssl sha1 | awk '{print $NF}') 
```

This also eliminates the need to conditionalize on uname.

File: pwned_pass.sh [hunk 2]
```diff
@@ -42,11 +47,19 @@ echo 'Looking up your password...'
...
+    count=$( echo "$response" \
+                 | grep -i "$hash_suffix" \
+                 | cut -d':' -f2 \
+                 | grep -Po '\d+' \
+                 || echo 0)
+elif [ $(uname) == "Darwin" ]; then
+    count=$( echo "$response" \
+                 | grep -i "$hash_suffix" \
+                 | cut -d':' -f2 \
+                 | egrep -o '\d+' \
```
Comment (codyps) [inline pwned_pass.sh]: `egrep` works on GNU systems too, just use that all the time.

**Comment by kevin-hanselman** [comment]
I like `grep -E` over `egrep`, but otherwise I'll merge @jmesmon's suggestions if they get Mac OS working.

**Comment by codyps** [comment]
Sure, `grep -E` should work too.

**Comment by crclark96** [comment]
`grep -E` isn't working on linux for me

**Comment by kevin-hanselman** [comment]
@crclark96 Is `grep -E` not working on Linux for you? If so, why did you commit it? It works on my Linux system.

Nonetheless, I'd be surprised if `grep -o '[0-9]\+'` doesn't work on both platforms.

**Comment by crclark96** [comment]
The previous commit only addressed the openssl difference. Both forms should work now, the correct regex sequence that works for both is `'[0-9]+'`

May have been a difference in system grep versions

**Comment by kevin-hanselman** [comment]
Thanks @crclark96 and @jmesmon!

**Pull request merged**